### DISCUSSION TITLE
Using Fontawesome with custom icons

### DISCUSSION THREAD

**Discussion by JenniferHem** [vuepress-theme-hope/vuepress-theme-hope#2682] [Q&A] [answered]
Hey @Mister-Hope,

amazing work on the Theme, we really love to use it for our docs!
We really would like to be able to use custom Icons for the actions section. So basically there should be a link to an application containg the applications logo. With my little knowledge I have I played around with fontawesome. One thin I realized was, that if you use the demo from your documentation and switch from iconfont to fontawesome some of the logos are not found, although they do exist as free fontawesome Icons. From there I tried to add the Icons via: 

```
import { library } from "@fortawesome/fontawesome-svg-core";
import { 
  faJs, faVuejs, faTwitter
} from '@fortawesome/free-brands-svg-icons';


library.add( faJs, faVuejs, faTwitter);
```
in the config.ts but the icons are not found. 

Now I am really clueless, honestly I understand too little about vuepress to decide wether I am just putting this in the wrong place, wether I just messed up the fontawesome setup or wether this is just currently not possible. 
Ultimately I would like to define my own Icon and add this to fontawesome via setting up my own IconDefinition. 

It would be great to get your feedback what I am doing wrong. Again its amazing how most of the thins just work outside the box and can be used with almost no knowledge about vuepress!

Thanks! Jennifer

**Comment by Mister-Hope** [comment]
I am not sure, but I am receiving more traffic than expected this week. You are supposed to use your own kit, maybe I should add it in docs.

![image](https://user-images.githubusercontent.com/33315834/212846718-2c9f6286-dd34-4851-a4ef-3b5ad7069980.png)

I will try to add fontawesome assets into theme.

**Reply by JenniferHem** [reply]
Thanks for the reply. Could you maybe elaborate on what you mean by "using your own kit"?

**Reply by Mister-Hope** [reply]
[fontawesome.com](https://fontawesome.com/kits)

**Reply by JenniferHem** [reply]
Ah okay, sorry. Unfortunately we cannot do this in our setting this is why I was hoping to do something along the lines of this tuorial [here](https://dev.to/astagi/add-custom-icons-to-font-awesome-4m67) but it seems this is not possible (at least not with my minimal knowledge)

**Reply by Mister-Hope** [reply]
We update fontawesome support.

**Reply by JenniferHem** [reply]
We will try, thanks a lot! 
so I suppose there is no way, apart from fontawesome kits that a custom icon can be added to the fontawesome library?

**Comment by Mister-Hope** [comment, marked as answer]
We improve icon support and this should solve your questions
